### PR TITLE
[DEV-4633] Take toptier agency into account when filtering by subtier

### PR DESCRIPTION
**Pull request by kbard**
**Description:**

We need to take the toptier agency into account when filtering by subtier since several subtier agency names span toptiers.

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
4. [x] Matview impact assessment completed
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] No Operations ticket required
8. [x] Jira Ticket [DEV-4633](https://federal-spending-transparency.atlassian.net/browse/DEV-4633):
    - [x] Link to this Pull-Request

### CODE REVIEW

File: usaspending_api/api_contracts/contracts/v2/autocomplete/awarding_agency.md [hunk 1]
```diff
@@ -30,7 +30,6 @@ This route sends a request to the backend to retrieve awarding agencies matching
 + `id` (required, number)
 + `toptier_flag` (required, boolean)
 + `toptier_agency` (required, object)
-    + `id` (required, number)
```
Comment (kbard) [inline usaspending_api/api_contracts/contracts/v2/autocomplete/awarding_agency.md]: This was something I added last sprint in preparation for this ticket.  Turns out I didn't need it so I removed it.

File: usaspending_api/database_scripts/matview_generator/mv_agency_autocomplete.json [hunk 2]
```diff
@@ -87,7 +87,7 @@
     "-- of them has been cited as an awarding agency, we need to ensure the final outcome is flagged as ",
     "-- an awarding agency). ",
     "select ",
-    "    min(agency_id) as agency_id, ",
+    "    min(agency_id) as agency_autocomplete_id, ",
```
Comment (kbard) [inline usaspending_api/database_scripts/matview_generator/mv_agency_autocomplete.json]: While derived from the agency_id, I didn't want anyone to think this was actual, legitimate agency data.  It is a derivation and is strictly designed to work with the autocomplete.  I did this with the TAS matview for the same reason but it slipped my mind when I created this matview.  As far as I can tell, nothing has latched onto the `agency_id` yet so this should affect nothing.

File: usaspending_api/common/query_with_filters.py [hunk 3]
```diff
@@ -119,7 +119,10 @@ def _generate_elasticsearch_query(cls, filter_values: List[dict], query_type: _Q
             agency_name = v["name"]
             agency_tier = v["tier"]
             agency_type = v["type"]
+            toptier_name = v.get("toptier_name")
             agency_query = ES_Q("match", **{f"{agency_type}_{agency_tier}_agency_name__keyword": agency_name})
+            if agency_tier == "subtier" and toptier_name is not None:
```
Comment (BrianZito) [inline usaspending_api/common/query_with_filters.py]:
```suggestion
            if agency_tier == "subtier" and toptier_name:
```

Comment (sethstoudenmier) [inline usaspending_api/common/query_with_filters.py]: Does this also need to be applied to `matview_filters.py` and `sub_award.py` where we filter for Postgres queries?

Comment (kbard) [reply usaspending_api/common/query_with_filters.py]: You mean [this](https://github.com/fedspendingtransparency/usaspending-api/blob/8790aea5c447bbfedffd86f5f840a63d1b94a8cf/usaspending_api/awards/v2/filters/matview_filters.py#L147-L152)?  Interestingly, it was already there.  I guess we had intended to support this functionality at one point... or maybe we did.  I don't know.  Feel free to double check it, although I didn't write it so it'll probably work.

Comment (sethstoudenmier) [reply usaspending_api/common/query_with_filters.py]: Yep that is it. Overlooked that when I checked.

**Comment by kbard** [comment]
> would be good to add some tests around the new conditional that you added

Yeah.  Good call.  Wanted to do that but it slipped my mind.  Too much going on.

**Pull request merged**